executor/coprocessor.go

```diff
@@ -0,0 +1,161 @@
+package executor
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/coprocessor"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/planner/core"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/timeutil"
+	"github.com/pingcap/tipb/go-tipb"
+)
+
+// HandleCopDAGRequest handles the coprocessor request.
+func HandleCopDAGRequest(ctx context.Context, sctx sessionctx.Context, req *coprocessor.Request) *coprocessor.Response {
```

Comment:
It's better to define a struct and make related functions as methods of the struct to avoid define too much function in the `executor` package scope. (e.g: `CoprocessorDAGHandler`)

Reply:
Great, Done.